todo-backend/README.adoc

```diff
@@ -0,0 +1,452 @@
+----
+$ oc new-app postgresql-ephemeral \
+   -p DATABASE_SERVICE_NAME=todos-db \
+   -p POSTGRESQL_DATABASE=todos
+----
+
+This will create a PostgreSQL database named `todos` on OpenShift that can be accessed on the port `5432` on the service `todos-db`.
+We don't need to copy the credentials to connect to the database as we will retrieve them later using the `todos-db` secret that was created
+when the database is deployed.
+
+ifndef::ProductRelease[]
+=== Build and Deploy the Backend on OpenShift with Bootable Jar
+
+The backend will be built and deployed on OpenShift with a Helm Chart for {productName}.
+
+[source,options="nowrap"]
+----
+$ helm install todo-backend -f todo-backend-bootable-jar.yaml wildfly/wildfly
+NAME: todo-backend
+...
+STATUS: deployed
+REVISION: 1
+----
+
+The Helm Chart for this quickstart contains all the information to build an image from the source code using Bootable Jar:
+
+[source,options="nowrap"]
+----
+build:
+  uri: https://github.com/wildfly/quickstart.git
+  mode: bootable-jar
+----
+endif::[]
+
+=== Build and Deploy the Backend on OpenShift with {productName} S2I
+
+The backend will be built and deployed on OpenShift with a Helm Chart for {productName}.
+
+ifndef::ProductRelease[]
+[source,options="nowrap"]
+----
+$ helm install todo-backend -f todo-backend-s2i.yaml wildfly/wildfly
+NAME: todo-backend
+...
+STATUS: deployed
+REVISION: 1
+----
+endif::[]
+ifdef::ProductRelease[]
+[source,options="nowrap"]
+----
+$ helm install todo-backend -f todo-backend-s2i.yaml jboss-eap/eap7
+NAME: todo-backend
+...
+STATUS: deployed
+REVISION: 1
+----
+endif::[]
+
+The Helm Chart for this quickstart contains all the information to build an image from the source code using S2I:
+
+ifndef::ProductRelease[]
+[source,options="nowrap"]
+----
+build:
+  uri: https://github.com/wildfly/quickstart.git
+  mode: s2i
+  s2i:
+    galleonLayers: cloud-server,postgresql-datasource,ejb
+----
+endif::[]
+ifdef::ProductRelease[]
+[source,options="nowrap"]
+----
+build:
+  uri: https://github.com/jboss-developer/jboss-eap-quickstarts.git
+  mode: s2i
+  s2i:
+    galleonLayers: cloud-server,postgresql-datasource,ejb
+----
+endif::[]
+
+=== Environment variables for PostgreSQL
+
+The Helm Chart also contains the environment variables required to connect to the PostgreSQL database.
+In local deployment the credentials were passed directly as the values of the environment variables.
+For OpenShift, we rely on secrets so that the credentials are never copied outside OpenShift:
+
+[source,options="nowrap"]
+----
+deploy:
+  env:
+    - name: POSTGRESQL_PASSWORD
+      valueFrom:
+        secretKeyRef:
+          key: database-password
+          name: todos-db
+----
+
+When the application is deployed, the value for the `POSTGRESQL_PASSWORD` will be taken from the key `database-password`
+in the secrets `todos-db`.
+
+Let's wait for the application to be built and deployed:
+
+[source,options="nowrap"]
+----
+$ oc get dc/todo-backend -w
```

Comment:
This fails for me:
```
$ oc get dc/todo-backend -w
Error from server (NotFound): deploymentconfigs.apps.openshift.io "todo-backend" not found
```

There is no `todo-backend` DeploymentConfig created, I can only see `todo-backend` Deployment instead
```
$ oc get deployment/todo-backend -w
NAME           READY   UP-TO-DATE   AVAILABLE   AGE
todo-backend   0/3     3            0           31s
todo-backend   1/3     3            1           4m31s
todo-backend   2/3     3            2           4m31s
todo-backend   3/3     3            3           4m31s
```

I am not sure why the difference and how you managed to get the output listed, we are using Openshift 4.7 for testing

I can also see similar output for ReplicaSet, but here the name has generated suffix, no easy way to determine it so it is probably not useful:
```
$ oc get rs/todo-backend-5b649477b7 -w
NAME                      DESIRED   CURRENT   READY   AGE
todo-backend-5b649477b7   3         3         3       2m3s
```

Reply:
good catch, the helm chart was using deployment config when I wrote the initial readme but it is not using a `deployment`. I used your output to fix it.